app/experimenter/nimbus-ui/src/components/FormBranches/FormBranch.tsx

```diff
@@ -3,30 +3,30 @@
  * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
 
 import React from "react";
-import { useFormContext, RegisterOptions, FieldError } from "react-hook-form";
+import { FieldError, useFormContext } from "react-hook-form";
 import Form from "react-bootstrap/Form";
 import Col from "react-bootstrap/Col";
 import Button from "react-bootstrap/Button";
 import Badge from "react-bootstrap/Badge";
 import { ReactComponent as DeleteIcon } from "../../images/x.svg";
 
 import { getExperiment_experimentBySlug } from "../../types/getExperiment";
-
 import {
   getConfig_nimbusConfig,
   getConfig_nimbusConfig_featureConfig,
 } from "../../types/getConfig";
 
 import { AnnotatedBranch } from "./reducer";
 import InlineErrorIcon from "../InlineErrorIcon";
+import { useCommonNestedForm } from "../../hooks";
 
-type FormBranchFields = {
-  name: string;
-  description: string;
-  ratio: string | number;
-  featureValue: string | null;
-  featureEnabled: boolean;
-};
+export const branchFieldNames = [
+  "name",
+  "description",
+  "ratio",
+  "featureValue",
+  "featureEnabled",
+] as const;
```

Comment:
I was going to suggest this could still just be a type (`type FormBranchFields = "name" | "description" | "ratio" | "featureValue" | "featureEnabled";`) but it looks like you're doing something with it in tests. Since the const is really only there to suit the tests you might copy it over there just for testing so we're not including code just for tests, but it's so minimal I don't feel strongly about it.

Reply:
I did it this way so we don't have to maintain the list of fieldnames in two places per form - it's done this way in `FormMetrics`, `FormAudience`, and `FormOverview` (granted I did it that way though, just continued the pattern here). I like that if you provide the wrong `name` to `FormErrors` or `formControlAttrs`, TS complains, and that we run over each one in tests to make sure `FormErrors` and `formControlAttrs` are called as expected.

Reply:
Works for me :)